### PR TITLE
Ohm 291 audit log redirect bug

### PR DESCRIPTION
**Pull request by BMartinos**
This PR relates to issue: #400 

There was a bug with how the location redirect URLs were constructed. When the Console was run in sub-domain, the location URL wouldn't be valid and redirect the user to the web server landing page. Same result were experienced in Apache.

A dependency upgrade also fixed bug: #359

### CODE REVIEW

File: app/scripts/controllers/audits.js [hunk 1]
```diff
@@ -407,12 +411,20 @@ angular.module('openhimConsoleApp')
...
-      $scope.refreshAuditsList();
+      // clear all filter parameters
+      $location.search({});
+
+      // get the filter params object after clearing them
+      var filterParamsAfterClear = JSON.stringify( angular.copy( $location.search() ) );
+
+      // if the filters object stays the same then call refresh function
+      // if filters object not the same then angular changes route and loads controller ( refresh )
+      if ( filterParamsBeforeClear === filterParamsAfterClear ){
```
Comment (rcrichton) [inline app/scripts/controllers/audits.js]: I think you can skip converting these to string and rather use `angular.equals` to do this check directly on the objects.

**Comment by rcrichton** [comment]
@BMartinos this looks good, nice catch. I'm still getting some redirects when entering invalid characters in the some of the date fields. Perhaps we should change this to work `onBlur` rather than right which seems to be `onChange`.

File: app/scripts/controllers/audits.js (hunk 1 above)
Comment (BMartinos) [inline app/scripts/controllers/audits.js]: Great works well!

**Comment by BMartinos** [comment]
Will have a look at whats causing the dates to redirect as well.

**Comment by BMartinos** [comment]
Hey @rcrichton Ive made the change as suggested and works well. regarding the change for the date, unfortunately ng-blur wouldn't have worked as its not triggered when using the mouse to click on a specific date. This then only works when typing the date. Ive added a function to check if the date is valid and only then run the apply filters function.

File: app/scripts/controllers/audits.js [hunk 2]
```diff
@@ -412,17 +419,19 @@ angular.module('openhimConsoleApp')
       $scope.settings.list.tabview = 'same';
 
       // get the filter params object before clearing them
-      var filterParamsBeforeClear = JSON.stringify( angular.copy( $location.search() ) );
+      //var filterParamsBeforeClear = JSON.stringify( angular.copy( $location.search() ) );
```
Comment (rcrichton) [inline app/scripts/controllers/audits.js]: Remove commented out code.

File: app/scripts/controllers/audits.js [hunk 3]
```diff
@@ -412,17 +419,19 @@ angular.module('openhimConsoleApp')
...
+      var filterParamsBeforeClear = angular.copy( $location.search() );
 
       // clear all filter parameters
       $location.search({});
 
       // get the filter params object after clearing them
-      var filterParamsAfterClear = JSON.stringify( angular.copy( $location.search() ) );
+      // var filterParamsAfterClear = JSON.stringify( angular.copy( $location.search() ) );
```
Comment (rcrichton) [inline app/scripts/controllers/audits.js]: remove commented out code

**Comment by rcrichton** [comment]
LGTM :+1:

**Comment by rcrichton** [comment]
Just remove the commented out code.

**Pull request merged**